server/src/main/java/org/elasticsearch/plugins/internal/DocumentSizeObserver.java

```diff
@@ -8,50 +8,39 @@
 
 package org.elasticsearch.plugins.internal;
 
+import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.xcontent.XContentParser;
 
 /**
  * An interface to allow wrapping an XContentParser and observe the events emitted while parsing
  * A default implementation returns a noop DocumentSizeObserver
  */
-public interface DocumentSizeObserver {
+public interface DocumentSizeObserver extends NormalizedDocumentSize {
```

Comment:
IDK, this looks weird to me, but maybe it's just the naming which always trips me (Observer, reporter, provider...)

What if, instead of inheriting, we add a method to return a `NormalizedDocumentSize`?

Reply:
I've done some renaming, but this is a never ending story. There's so much more and I don't want to explode the scope here.
`DocumentSizeObserver` become a `XContentMeteringParserDecorator` that exposes `DocumentSize meteredDocumentSize()`.
Let me know what you think @ldematte